### PR TITLE
scheduler: move init function

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref #5837

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by lhy1024** [comment]
cc @rleungx @binshi-bing

**Review by binshi-bing — APPROVED** [review]
LGTM. Other alternatives might be [Package plugin](https://pkg.go.dev/plugin) or k8s sidecar.

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -136,6 +135,7 @@ func createServerWrapper(args []string) (context.Context, context.CancelFunc, ba
 	if err != nil {
 		log.Fatal("create server failed", errs.ZapError(err))
 	}
+	schedulers.Register()
```
Comment (binshi-bing) [inline cmd/pd-server/main.go:138]: Other alternatives might be [Package plugin](https://pkg.go.dev/plugin) or k8s sidecar.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**